### PR TITLE
ASC control refactor changes

### PR DESCRIPTION
**Pull request by ritika-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json [hunk 1]
```diff
@@ -546,6 +510,114 @@
...
+        "Automated",
+        "Config",
+        "SOX",
+        "SubscriptionCore"
+      ],
+      "Enabled": true,
+      "DataObjectProperties": [
+        "id",
+        "properties.logCollection",
+        "properties.recommendations",
```
Comment (v-zhgup) [inline src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json]: check what all recommendations are captured in attested state and if they are actually required for the control attestation

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 2]
```diff
@@ -1943,5 +1914,67 @@ class SubscriptionCore: AzSVTBase
...
+	hidden [ControlResult] CheckSecurityPolicy([ControlResult] $controlResult)
+	{
+		if ($this.SecurityCenterInstance)
+		{
+			$this.SubscriptionContext.SubscriptionMetadata.Add("MissingOptionalASCPolicies",$this.MisConfiguredOptionalASCPolicies);
+			$this.SubscriptionContext.SubscriptionMetadata.Add("MissingMandatorySecurityPolicies",$this.MisConfiguredSecurityPolicySettings);
+
+			if(($this.MisConfiguredSecurityPolicySettings | Measure-Object).Count -ne 0)
+			{
+				$controlResult.EnableFixControl = $true;
```
Comment (v-zhgup) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Break down the fix script in accordance to the controls being categorized

File: src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json [hunk 3]
```diff
@@ -546,6 +510,114 @@
       ],
       "Enabled": true,     
       "Rationale": "The regular / day to day use accounts are subject to a lot of credential theft attacks due to various activities that a user conducts using such accounts (e.g., browsing the web, clicking on email links, etc.). A user account that gets compromised (say via a phishing attack) immediately subjects the entire cloud subscription to risk if it is a member of critical roles in the subscription. Use of smartcard-backed alternate (SC-ALT) accounts instead protects the cloud subscriptions from this risk. Moreover, for complete protection, all sensitive access must be done using a secure admin workstation (SAW) and Azure Privileged Identity Management (PIM)."
+    },
+    {
+      "ControlID": "Azure_Subscription_Config_ASC_Security_Policy",
+      "Description": "Azure Security Center (ASC) policies must be correctly configured on the subscription.",
+      "Id": "SubscriptionCore330",
+      "ControlSeverity": "Medium",
```
Comment (abhaydaga) [inline src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json]: Shouldn't the severity be High here?

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 4]
```diff
@@ -1943,5 +1914,67 @@ class SubscriptionCore: AzSVTBase
...
+	hidden [ControlResult] CheckAutoProvisioningForSecurity([ControlResult] $controlResult)
+	{
+		if ($this.SecurityCenterInstance)
+		{
+			$this.SubscriptionContext.SubscriptionMetadata.Add("MissingAutoProvisioningPolicies",$this.MisConfiguredAutoProvisioningSettings);
+
+			if(-not [string]::IsNullOrWhiteSpace($this.MisConfiguredAutoProvisioningSettings))
+		      {
+			     $controlResult.SetStateData("Misconfigured AutoProvisioning Policy", $this.MisConfiguredAutoProvisioningSettings);
+			     $controlResult.AddMessage([VerificationResult]::Failed, [MessageData]::new("AutoProvisioning setting is not configured. Please update the setting in order to comply", $this.MisConfiguredAutoProvisioningSettings));
```
Comment (abhaydaga) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: AutoProvisioning setting is disabled.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 5]
```diff
@@ -1943,5 +1914,67 @@ class SubscriptionCore: AzSVTBase
...
+	hidden [ControlResult] CheckSecurityContactDetails([ControlResult] $controlResult)
+	{
+		if ($this.SecurityCenterInstance)
+		{
+			$this.SubscriptionContext.SubscriptionMetadata.Add("MissingSecurityContactDetails",$this.MisConfiguredSecurityContactDetails);
+
+			if(-not [string]::IsNullOrWhiteSpace($this.MisConfiguredSecurityContactDetails))
+		      {
+			     $controlResult.SetStateData("Misconfigured Security Contact Details", $this.MisConfiguredSecurityContactDetails);
+			     $controlResult.AddMessage([VerificationResult]::Failed, [MessageData]::new("Security Contacts are not configured. Please update the setting in order to comply", $this.MisConfiguredSecurityContactDetails));
```
Comment (abhaydaga) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: contacts

Please update the setting in order to comply - remove this

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 2 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: added new fix control as required. Changes pushed in latest commit

File: src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json (hunk 3 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json]: made the changes in latest commit

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 4 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Made the changes

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 5 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Made the changes

File: src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json (hunk 1 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Configurations/SVT/SubscriptionCore/SubscriptionCore.json]: Checked the blob, only the data that is being captured using setdata is stored. Shared the blob with you as well. Plz let me know if any changes are required

**Pull request merged**